aten/src/ATen/native/sparse/SoftMax.cpp

```diff
@@ -505,7 +506,102 @@ void cpu_sparse_coo_softmax_backward(Tensor& grad_input, const Tensor& grad, con
 
 } // namespace
 
-Tensor softmax_sparse_cpu(const Tensor& input_, const int64_t dim_, const bool half_to_float) {
+void softmax_sparse_cpu(
```

Comment:
Maybe not in this PR, but should there eventually be native/sparse/cpu, analogous to aten/native/cpu vs aten/native/cuda?